test/parallel/test-querystring.js

```diff
@@ -51,6 +51,8 @@ const qsTestCases = [
      __defineGetter__: 'baz' }],
   // See: https://github.com/joyent/node/issues/3058
   ['foo&bar=baz', 'foo=&bar=baz', { foo: '', bar: 'baz' }],
+  ['&&foo=bar&&', 'foo=bar', { foo: 'bar' }],
+  ['&&&&', '', {}],
```

Comment:
That's right! Adding the cases that the key or the value is empty sounds useful for revealing the current behaviour. I've added them :)